tests/moduletool/test_install.py

```diff
@@ -941,3 +942,218 @@ def test_install_project_with_install_mode_master(tmpdir: py.path.local, snippet
+            os.path.join(str(tmpdir), "parent_module"),
+            new_version=version.Version("1.2.3"),
+            new_name="parent_module",
+            install=True,
+            new_requirements=[InmantaModuleRequirement.parse("child_module==3.3.3")],
+        )
+
+    log_contains(
+        caplog,
+        "inmanta.env",
+        logging.DEBUG,
+        "No matching distribution found for inmanta-module-child-module==3.3.3",
+    )
+
+    shutil.rmtree(os.path.join(str(tmpdir), "parent_module"), ignore_errors=True)
+    shutil.rmtree(os.path.join(str(tmpdir), "child_module"), ignore_errors=True)
+    # Scenario 3
+    # parent_module requires child_module v3.3.3
+    # child_module is installed with v3.3.3
+
+    # Install the required module with the correct version:
+    module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "child_module"),
+        new_version=version.Version("3.3.3"),
+        new_name="child_module",
+        install=True,
+    )
+
+    module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "parent_module"),
+        new_version=version.Version("1.2.3"),
+        new_name="parent_module",
+        install=True,
+        new_requirements=[InmantaModuleRequirement.parse("child_module==3.3.3")],
+    )
+
+    log_contains(
+        caplog,
+        "inmanta.env",
+        logging.DEBUG,
+        "Successfully installed inmanta-module-parent-module-1.2.3",
+    )
+
+
+def test_pip_output(local_module_package_index: str, snippetcompiler_clean, caplog, modules_v2_dir, tmpdir):
+    """
+    This test checks that pip's output is correctly logged on module install.
+    """
+    caplog.set_level(logging.DEBUG)
+
+    snippetcompiler_clean.setup_for_snippet("")
+
+    modone: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "modone"),
+        new_version=version.Version("3.1.2"),
+        new_name="modone",
+        install=True,
+    )
+    modtwo: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "modtwo"),
+        new_version=version.Version("2.2.2"),
+        new_name="modtwo",
+        new_requirements=[InmantaModuleRequirement.parse("modone")],
+        install=True,
+    )
+
+    modules = ["modone", "modtwo"]
+    v2_requirements = [Requirement.parse(module.ModuleV2Source.get_package_name_for(mod)) for mod in modules]
+
+    snippetcompiler_clean.setup_for_snippet(
+        f"""
+        import {module.ModuleV2.get_name_from_metadata(modone)}
+        import {module.ModuleV2.get_name_from_metadata(modtwo)}
+        """,
+        autostd=False,
+        python_package_sources=[local_module_package_index],
+        python_requires=v2_requirements,
+        install_project=True,
+    )
```

Comment:
Fixed this with changing `install_project=True,` to `False`.
Out of curiosity I tried the other way round (By changing install to False in both `module_from_template` and keeping install_project to True) but I couldn't make it work, what am I missing?

Reply:
It doesn't work the other way around because the modules, generated from the templates, are not stored in a python index. As such the install on the project can't find the requested packages. That makes me realize it's probably better to use the install from the `setup_for_snippet` everywhere instead of the install from the `module_from_template`. The former is a simulation of the `inmanta project install` command while the latter is one of the `inmanta module install` command. This debugging information is mainly relevant for the `inmanta project install` command, so it's probably better for the tests to also use that logic.